### DISCUSSION TITLE
Can I use DDEV with self-hosted cloud development tools like Coder or DevPod?

### DISCUSSION THREAD

**Discussion by otowa1** [ddev/ddev#5708] [General]
I want to use them because they are tools like Codespaces that are not vendor-locked-in, reduce costs, and do not pollute the local environment, but I am stumped by the domain setup.
Is there any way to get around this?

**Comment by tyler36** [comment]
I haven't heard of anyone doing trying (or even Coder or DevPod).  

Take at look at the Codespaces PR: https://github.com/ddev/ddev/pull/4476
If I recall, DDEV disables the router for both Codespaces and Gitpod.

**Reply by otowa1** [reply]
Yes, routing seems to be the issue.

**Reply by tyler36** [reply]
You could try making a custom build of DDEV with changes similar to the codespace PR above.
https://ddev.readthedocs.io/en/latest/developers/building-contributing/

There was a recent training session which might help you get setup.
https://www.dropbox.com/scl/fi/up13ll31115zo997kqzac/2023-08-15_contributor_training_maintain_ddev.com.mp4?rlkey=uff5jra0dp524utuigv9j6399&dl=0

**Reply by otowa1** [reply]
Thanks.I will try it.

**Comment by tyler36** [comment]
@otowa552 Someone was working on DDEV with Devpod today in Discord. 
Did you get it wokring? Do you a working repo or blog site that helped you?

**Reply by otowa1** [reply]
No, I could not get it to work.

**Reply by rfay** [reply]
The devpod discussion in Discord was https://discord.com/channels/664580571770388500/1293897859242004541

If you need an invite to Discord, it's https://ddev.com/s/discord
